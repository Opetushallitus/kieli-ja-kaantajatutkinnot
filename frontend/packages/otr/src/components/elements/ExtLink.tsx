import { ButtonProps } from '@mui/material';
import { FC } from 'react';

import { CustomButton } from 'components/elements/CustomButton';
import { Color } from 'enums/app';

export interface ExtLinkProps {
  text: string;
  href: string;
}

export const ExtLink: FC<ButtonProps & ExtLinkProps> = ({
  text,
  href,
  endIcon,
  className,
  'aria-label': ariaLabel,
}) => {
  return (
    <CustomButton
      className={className}
      target="_blank"
      rel="noreferrer"
      color={Color.Inherit}
      href={href}
      endIcon={endIcon}
      aria-label={ariaLabel}
    >
      {text}
    </CustomButton>
  );
};
