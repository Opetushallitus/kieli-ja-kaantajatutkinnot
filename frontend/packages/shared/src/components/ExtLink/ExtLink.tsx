import { ButtonProps } from '@mui/material';
import { FC } from 'react';

import { Color } from '../../enums/common';
import { CustomButton } from '../CustomButton/CustomButton';

export interface ExtLinkProps {
  text: string;
  href: string;
}

export const ExtLink: FC<ButtonProps & ExtLinkProps> = ({
  text,
  href,
  startIcon,
  endIcon,
  className,
  'aria-label': ariaLabel,
  ...rest
}) => {
  return (
    <CustomButton
      className={className}
      target="_blank"
      rel="noreferrer"
      color={Color.Inherit}
      href={href}
      startIcon={startIcon}
      endIcon={endIcon}
      aria-label={ariaLabel}
      {...rest}
    >
      {text}
    </CustomButton>
  );
};
