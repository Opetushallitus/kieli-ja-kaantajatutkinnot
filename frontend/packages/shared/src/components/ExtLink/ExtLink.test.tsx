import renderer from 'react-test-renderer';

import { ExtLink } from './ExtLink';

describe('ExtLink', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<ExtLink text="Test link" href="https://someothersite.com" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
