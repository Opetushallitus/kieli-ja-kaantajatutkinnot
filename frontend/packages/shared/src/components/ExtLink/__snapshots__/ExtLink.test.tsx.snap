// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtLink should render correctly 1`] = `
<a
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
  href="https://someothersite.com"
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  rel="noreferrer"
  tabIndex={0}
  target="_blank"
>
  Test link
</a>
`;
